w3af/plugins/tests/grep/test_meta_generator.py

```diff
@@ -0,0 +1,122 @@
+"""
+test_meta_generator.py
+
+Copyright 2012 Andres Riancho
+
+This file is part of w3af, http://w3af.org/ .
+
+w3af is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation version 2 of the License.
+
+w3af is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with w3af; if not, write to the Free Software
+Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+"""
+import unittest
+from itertools import repeat
+
+from mock import patch
+
+import w3af.core.data.constants.severity as severity
+import w3af.core.data.kb.knowledge_base as kb
+from w3af.core.controllers.ci.moth import get_moth_http
+from w3af.core.data.url.HTTPResponse import HTTPResponse
+from w3af.core.data.request.fuzzable_request import FuzzableRequest
+from w3af.core.data.parsers.doc.url import URL
+from w3af.core.data.dc.headers import Headers
+from w3af.core.controllers.misc.temp_dir import create_temp_dir
+from w3af.plugins.tests.helper import PluginTest, PluginConfig
+from w3af.plugins.grep.meta_generator import meta_generator
+
+
+class TestMetaGenerator(PluginTest):
+
+    meta_generator_url = get_moth_http('/grep/meta_generator/')
```

Comment:
Did you make any modifications to moth? Will this test PASS without them?

Reply:
i did not, no. as such this test might _fail_. hmm ...